src/libServer/Server.cpp

```diff
@@ -1106,3 +1106,49 @@ string Server::GetNumTxnsDSEpoch() {
     return "0";
   }
 }
+
+Json::Value Server::GetTransactionsForTxBlock(const string& txBlockNum,
+                                              unsigned int shardID) {
+  LOG_MARKER();
+  LOG_GENERAL(INFO, txBlockNum << " " << shardID);
+  uint64_t txNum;
+  Json::Value _json = Json::arrayValue;
+  try {
+    txNum = strtoull(txBlockNum.c_str(), NULL, 0);
+  } catch (exception& e) {
+    throw JsonRpcException(RPC_INVALID_PARAMETER, e.what());
+  }
+
+  auto const& txBlock = m_mediator.m_txBlockChain.GetBlock(txNum);
+
+  if (txBlock == TxBlock()) {
+    throw JsonRpcException(RPC_INVALID_PARAMS, "Tx Block does not exist");
+  }
+
+  auto microBlockInfos = txBlock.GetMicroBlockInfos();
+
+  for (auto const& mbInfo : microBlockInfos) {
+    MicroBlockSharedPtr mbptr;
+    if (mbInfo.m_txnRootHash == TxnHash() && mbInfo.m_shardId == shardID) {
+      throw JsonRpcException(RPC_INVALID_PARAMETER, "Shard has no txns empty");
```

Comment:
Seems like `RPC_MISC_ERROR` is a better code? Also, maybe can just say "Shard has no txns"

Reply:
yeah i think i will replace it with database error